confluent_kafka/avro/serializer/message_serializer.py

```diff
@@ -213,4 +249,7 @@ def decode_message(self, message):
             if magic != MAGIC_BYTE:
                 raise SerializerError("message does not start with magic byte")
             decoder_func = self._get_decoder_func(schema_id, payload)
-            return decoder_func(payload)
+            return _wrap(
```

Comment:
Wouldn't it be better that the `_wrap` call is moved inside the `decoder` closure of `_get_decoder_func`?
Then you don't have to do the `self.registry_client.get_by_id(schema_id)` calls on each record because the schema is already available at construction time of the decoder function inside `_get_decoder_func`.